app/src/main/java/com/asfoundation/wallet/promotions/voucher/SkuButtonsAdapter.kt

```diff
@@ -0,0 +1,101 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import android.content.Context
+import android.graphics.Rect
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AbsListView
+import android.widget.Button
+import android.widget.GridView
+import androidx.recyclerview.widget.RecyclerView
+import com.asf.wallet.R
+import io.reactivex.subjects.PublishSubject
+import java.util.concurrent.atomic.AtomicReference
+
+
+class SkuButtonsAdapter(
+    val context: Context,
+    val buttonModels: List<SkuButtonModel>,
+    val onSkuClick: PublishSubject<Int>
+) :
+    RecyclerView.Adapter<SkuButtonsViewHolder>() {
+
+  private var inflater: LayoutInflater
+  private var activatedButton: AtomicReference<Button?> = AtomicReference()
+  private var selectedPosition: Int = -1
+
+  init {
+    inflater = LayoutInflater.from(context)
+  }
+
+  override fun onBindViewHolder(holder: SkuButtonsViewHolder, position: Int) {
+    holder.bind(position, selectedPosition, buttonModels.get(position), activatedButton, onSkuClick)
+  }
+
+  override fun getItemCount(): Int {
+    return buttonModels.size
+  }
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SkuButtonsViewHolder {
+    val button: Button =
+        inflater.inflate(R.layout.e_voucher_details_diamonds_button, null) as Button
+    val factor: Float =
+        parent?.getContext()
+            ?.getResources()
+            ?.getDisplayMetrics()?.density ?: 0.0.toFloat()
+    button.layoutParams = AbsListView.LayoutParams(GridView.AUTO_FIT, (factor * 48).toInt())
+
+    return SkuButtonsViewHolder(button)
+  }
+
+  fun setSelectedSku(index: Int) {
+    selectedPosition = index
+  }
+
+  fun getSelectedSku(): SkuButtonModel = buttonModels.get(selectedPosition)
+}
+
+class SkuButtonsViewHolder(private val button: Button) : RecyclerView.ViewHolder(button) {
```

Comment:
I don't mind having the ViewHolder in the same file as the Adapter but I don't think we do that in the project. I think we generally separate into a new file.